src/scripts/main.js

```diff
@@ -1 +1,75 @@
 'use strict';
+
+const notificationDiv = document.querySelector('[data-qa="notification"]');
+const body = document.querySelector('body');
+
+let leftClicked = false;
+let rightClicked = false;
+
+const firstPromise = new Promise((resolve, reject) => {
+  const clickHandler = (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved on a left click in the document');
+    }
+  };
+
+  body.addEventListener('click', clickHandler, { once: true });
+
+  setTimeout(() => {
+    body.removeEventListener('click', clickHandler);
+    reject(new Error('First promise was rejected in 3 seconds if not clicked'));
+  }, 3000);
+});
+
+firstPromise.then(
+  (message) => showNotification('success', message),
+  (error) => showNotification('warning', error.message)
```

Comment:
FYI: it works if you explicitly `reject` in the promise, but will not work if some error is thrown unexpectedly - for this purpose you need `catch`